salt/utils/master.py

```diff
@@ -146,6 +146,8 @@ def _get_cached_minion_data(self, *minion_ids):
             if not salt.utils.verify.valid_id(self.opts, minion_id):
                 continue
             mdata = self.cache.fetch('minions/{0}'.format(minion_id), 'data')
+            if not isinstance(mdata, dict):
```

Comment:
Which cache is returning `None`? AFAIK caches should always be returning dictionaries even if they are empty.